src/app/services/catapultOptin.service.js

```diff
@@ -165,6 +168,50 @@ class CatapultOptin {
         });
     }
 
+    /**
+     * Signs and Sends DTOS via Ledger
+     *
+     * @param common
+     * @param dtos
+     */
+    _sendLedgerDTOs(common, dtos) {
+        return new Promise((resolve, reject) => {
+            Promise.all( dtos.map( dto => this.createTransactionFromDTO(dto, common))).then( transactions => {
+                const signTransaction = (i) => {
+                    return this._Ledger.serialize(transactions[i], this._Wallet.currentAccount, true).then( serialized => {
+                        if (transactions.length - 1 === i) {
+                            return [JSON.stringify(serialized)];
+                        }
+                        return new Promise(resolve => setTimeout( () => signTransaction(i + 1).then((next) => {
```

Comment:
could you elaborate on why set timer here?

Reply:
Removed timer